src/core/crypto/secp256k1.ts

```diff
@@ -124,6 +128,36 @@ export class Secp256k1PublicKey extends PublicKey {
   static isInstance(publicKey: PublicKey): publicKey is Secp256k1PublicKey {
     return "key" in publicKey && (publicKey.key as any)?.data?.length === Secp256k1PublicKey.LENGTH;
   }
+
+  /**
+   * Recover a Secp256k1 public key from a signature and message.
+   *
+   * @param args - The arguments for recovering the public key.
+   * @param args.signature - The signature to recover the public key from.
+   * @param args.message - The message that was signed.
+   * @param args.recoveryBit - The recovery bit to use for the public key.
+   */
+  static fromSignatureAndMessage(args: {
+    signature: HexInput | Secp256k1Signature;
+    message: HexInput;
+    recoveryBit: number;
```

Comment:
I would make the recoveryBit optional at this level, and handle it with recursion.

```typescript
static fromSignedMessage({ signature, message, recoveryBit }) {
  if (recoveryBit === undefined) {
    try {
      for (let i = 0; i < 4; i += 1) {
        return Secp256k1PublicKey.fromSignedMessage({ signature, message recoveryBit: i });
      }
    } catch {
      // ...
    }
    throw new Error('Unable to derive public key');
  }
  
  // regular implementation
}
```

Reply:
or have a private internal implementation I guess

Reply:
Recovery bit must be set here, because if you loopthrough all options you actually get two public keys.